algorithms/src/snark/varuna/varuna.rs

```diff
@@ -525,50 +530,39 @@ where
             .map(|c| proof::WitnessCommitments { w: *c.commitment() })
             .collect_vec();
         let fourth_commitments_chunked = fourth_commitments.chunks_exact(3);
+        let (g_a_commitments, g_b_commitments, g_c_commitments) = fourth_commitments_chunked
+            .map(|c| (*c[0].commitment(), *c[1].commitment(), *c[2].commitment()))
+            .multiunzip();
 
         #[rustfmt::skip]
         let commitments = proof::Commitments {
             witness_commitments,
             mask_poly,
-
             h_0: *second_commitments[0].commitment(),
-
             g_1: *third_commitments[0].commitment(),
             h_1: *third_commitments[1].commitment(),
-
-            g_a_commitments: fourth_commitments_chunked.clone().map(|c| *c[0].commitment()).collect(),
-            g_b_commitments: fourth_commitments_chunked.clone().map(|c| *c[1].commitment()).collect(),
-            g_c_commitments: fourth_commitments_chunked.map(|c| *c[2].commitment()).collect(),
-
+            g_a_commitments,
+            g_b_commitments,
+            g_c_commitments,
             h_2: *fifth_commitments[0].commitment(),
         };
 
-        let labeled_commitments: Vec<_> = circuit_commitments
-            .into_iter()
-            .flatten()
-            .zip_eq(AHPForR1CS::<E::Fr, MM>::index_polynomial_info(circuit_ids.iter()).values())
-            .map(|(c, info)| LabeledCommitment::new_with_info(info, *c))
-            .chain(first_commitments.into_iter())
-            .chain(second_commitments.into_iter())
-            .chain(third_commitments.into_iter())
-            .chain(fourth_commitments.into_iter())
-            .chain(fifth_commitments.into_iter())
-            .collect();
-
         // Gather commitment randomness together.
-        let commitment_randomnesses: Vec<Randomness<E>> = keys_to_constraints
-            .keys()
-            .flat_map(|pk| pk.circuit_commitment_randomness.clone())
+        let indexer_randomness = vec![Randomness::<E>::empty(); 6 * num_unique_circuits];
+        let commitment_randomnesses: Vec<Randomness<E>> = indexer_randomness
+            .into_iter()
             .chain(first_commitment_randomnesses)
             .chain(second_commitment_randomnesses)
             .chain(third_commitment_randomnesses)
             .chain(fourth_commitment_randomnesses)
             .chain(fifth_commitment_randomnesses)
             .collect();
 
+        let empty_randomness = Randomness::<E>::empty();
         if !MM::ZK {
-            let empty_randomness = Randomness::<E>::empty();
             assert!(commitment_randomnesses.iter().all(|r| r == &empty_randomness));
+        } else {
+            assert!(commitment_randomnesses.iter().any(|r| r != &empty_randomness));
```

Comment:
Let's switch the cases so that we can say `if MM::ZK` (avoiding the negation)

Reply:
Done 68f294cea